jaxrs-api/src/test/java/javax/ws/rs/core/EntityTagTest.java

```diff
@@ -0,0 +1,46 @@
+package javax.ws.rs.core;
+
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import javax.ws.rs.core.testrules.RuntimeDelegateTestRule;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+public class EntityTagTest {
+    private EntityTag tag1;
+    private EntityTag tag2;
+    private EntityTag tag3;
+    private EntityTag tag4;
+
+    @ClassRule
+    public static RuntimeDelegateTestRule runtimeDelegateTestRule = new RuntimeDelegateTestRule();
+
+    @Before
+    public void setUp() {
+        tag1 = new EntityTag("value1", true);
+        tag2 = new EntityTag("value1", true);
+        tag3 = new EntityTag("value2", true);
+        tag4 = new EntityTag("value1", false);
+    }
+
+    @Test
+    public void testEqualsAndHashCode() {
```

Comment:
Would be great if the test would be more self-describing (like BDD style), so reviewers can more easily understand what the target of the actual test is. Currently this test just says it does test methods.

Reply:
I am happy do change this. But looking at the other tests the test style seems to be the "traditional" junit test methods. So I didn't want to introduce a new style for only one test class.

If we want to use a "BDD style" we probably should refactor all tests.

Reply:
Refactoring sounds good since BDD is much more readable, but as it is a very heavy change, one should discuss the idea on the dev list. For now, I would be happy with writing all _new_ code in BDD style.